src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,44 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        if (type == Bulldozer.class) {
+            return new ArrayList<>(new BulldozerProducer().get());
+        } else if (type == Truck.class) {
+            return new ArrayList<>(new TruckProducer().get());
+        } else if (type == Excavator.class) {
+            return new ArrayList<>(new ExcavatorProducer().get());
+        } else {
+            return new ArrayList<>();
+        }
+    }
+
+    @Override
+    public void fill(List<? super Machine> machines, Machine value) {
+        for (int i = 0; i < machines.size(); i++) {
+            machines.set(i, value);
+        }
+    }
+
+    @Override
+    public void startWorking(List<? extends Machine> machines) {
+        if (machines.get(0).getClass() == Bulldozer.class) {
+            new Bulldozer().doWork();
+        } else if (machines.get(0).getClass() == Excavator.class) {
+            new Excavator().doWork();
+        } else {
+            new Truck().doWork();
+        }
```

Comment:
You should not check the class
```suggestion
        for (Machine machine: machines) {
            machine.doWork();
        }
```